### DISCUSSION TITLE
Theme designer / Code Editor

### DISCUSSION THREAD

**Discussion by The-Real-1** [primefaces/community#25] [PrimeNG Templates] [answered]
Hi,

I have a question about the roadmap of the Theme designer / Code Editor.
I have seen this note about the incomming architecture : 

```
Note: In upcoming versions, theming architecture will be redesigned to utilize CSS variables instead of SCSS variables in a backward compatible way for a dynamic approach. In addition, a new Unstyled mode will be provided as an alternative to the default styling so that CSS libraries like Tailwind or Bootstrap can be used to style the components. This work is planned to be completed in Q3 2023.
```

Will this be a breaking change or a compatibility layer will be available ?

Another question : 

Is it planned to migrate from Sass @import to @use in further releases ?

Thanks a lot,

Regards

**Comment by cagataycivici** [comment, marked as answer]
The implementation is in POC state so cannot say if it will be backward compatible or not but I think we have to make it backward compatible so old theme.css files can still be used. I'll have more after friday's meeting where we'll go over the POC. The DOM structure does not change so older theme files should still work to keep backward compatibility until migration.

About use vs import, we won't be using SASS anymore for theming of UI libraries. For example, primevue-sass-theme code will move into primevue and the SCSS variables there will be replaced as CSS variables.

**Reply by nhoizey** [reply]
Is it true for PrimeNG too?

(The question was first asked by @The-Real-1 in https://github.com/primefaces/primeng-sass-theme/issues/43 )

**Reply by SoyDiego** [reply]
If i'm not wrong, yes.

**Reply by The-Real-1** [reply]
@cagataycivici Thanks for the information.

**Comment by octavianvoloaca-tremend** [comment]
Are there any updates regarding the release of the CSS variables based version?

**Comment by autonomobil** [comment]
Any updates?

**Comment by cetincakiroglu** [comment]
Hi @octavianvoloaca-tremend @autonomobil ,

There isn't an exact date. It's in development.
